### PR TITLE
Use (( merge )) as the default behavior

### PR DESCRIPTION
**Pull request by filefrog**

### CODE REVIEW

**Comment by filefrog** [comment]
For #28

File: merge.go [hunk 1]
```diff
@@ -217,3 +249,31 @@ func shouldKeyMergeArray(obj []interface{}) (bool, string) {
...
+
+		obj := o.(map[interface{}]interface{})
+		if _, ok := obj[key]; !ok {
+			return fmt.Errorf("%s.%d: %s object does not contain the key '%s' - cannot merge", node, i, disp, key)
+		}
+	}
+	return nil
+}
+
+func shouldReplaceArray(obj []interface{}) bool {
```
Comment (geofffranks) [inline merge.go]: We should have unit tests for this function as well, to verify the regexes get matched right.. should be pretty copy/paste-able from the other shouldXXXX tests

File: README.md [hunk 2]
```diff
@@ -58,11 +58,11 @@ Arrays can be merged in three ways - prepending data, appending data, and comple
   - (( prepend ))
   ```
 
-- To merge the two arrays together (each index of the new array will be merged into the original, additionals appended),
+- To replace the first array with the second,
   ensure that the first element in the new array is <br>
 
   ```yml
-  - (( inline ))
+  - (( replace ))
```
Comment (geofffranks) [inline README.md]: looks like you dropped the docs for (( inline )), but inline is still supported as an explicit thing?

File: README.md [hunk 3]
```diff
@@ -82,7 +82,9 @@ Arrays can be merged in three ways - prepending data, appending data, and comple
   like objects in the array elements. The second is used to customize which key to use. See [Merging Arrays of Maps](#mapmerge)
   for an example.
 
-- To completely replace the array, don't do anything special - just make the new array what you want it to be!
+- If you don't specify a specific merge strategy, the array will
+  be merged automatically; using keys if they exist, and array
+  indices otherwise.
```
Comment (geofffranks) [inline README.md]: An example + linking to the example would probably be good here

File: merge.go [hunk 4]
```diff
@@ -80,97 +80,128 @@ func mergeArray(orig []interface{}, n []interface{}, node string) ([]interface{}
...
+		DEBUG("%s: performing explicit inline array merge", node)
+		var err error
+		merged, err = mergeArrayInline(orig, n[1:], node)
+		if err != nil {
+			return nil, err
+		}
+
+	} else if shouldReplaceArray(n) {
+		DEBUG("%s: replacing with new data (no specific array merge behavior requested)", node)
+		merged = n
```
Comment (geofffranks) [inline merge.go]: This code is never reached by tests

File: merge.go [hunk 5]
```diff
@@ -80,97 +80,128 @@ func mergeArray(orig []interface{}, n []interface{}, node string) ([]interface{}
...
+			return nil, err
+		}
+	} else {
+		DEBUG("%s: performing index-based array merge", node)
+		var err error
+
+		merged = nil
+		if err = canKeyMergeArray("original", orig, node, "name"); err == nil {
+			if err = canKeyMergeArray("new", n, node, "name"); err == nil {
+				merged, err = mergeArrayByKey(orig, n, node, "name")
```
Comment (geofffranks) [inline merge.go]: The err returned by this is never tested

File: merge.go [hunk 6]
```diff
@@ -80,97 +80,128 @@ func mergeArray(orig []interface{}, n []interface{}, node string) ([]interface{}
...
 			}
-			last = i
 		}
 
-		if len(orig) > 0 {
-			last++ // move to next index after finishing the orig slice - but only if we looped
+		if merged == nil {
+			merged, err = mergeArrayInline(orig, n, node)
+			if err != nil {
+				return nil, err
```
Comment (geofffranks) [inline merge.go]: This code is never reached by tests

**Pull request merged**